core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1219,6 +1219,9 @@ class Log(@volatile private var _dir: File,
               appendInfo.logAppendTime = duplicate.timestamp
               appendInfo.logStartOffset = logStartOffset
             case None =>
+              if (logDirFailureChannel.logDirIsOffline(parentDir)) {
+                throw new KafkaStorageException(s"The log dir $parentDir has failed.");
```

Comment:
Maybe we can leave a more informative error message here? Sth like "... dir has failed due to a previous IO exception", just indicating it is not failed because of the current calling trace.

Reply:
@guozhangwang good point. I also changed the language to 'offline' rather than 'failed', since this is more consistent with other usage.